whirlpool/eventsocket.py

```diff
@@ -45,49 +61,95 @@ async def _recv_msg(self, websocket: aiohttp.ClientWebSocketResponse):
+        timeout = aiohttp.ClientTimeout(total=None, connect=60, sock_connect=60)
+
+        try:
+            async with aiohttp.ClientSession(timeout=timeout) as session:
+                LOGGER.debug(f"Connecting to {self._url}")
+                async with session.ws_connect(
+                    self._url,
+                    timeout=timeout,
+                    autoclose=True,
+                    autoping=True,
+                    heartbeat=45,
+                ) as ws:
+                    self._websocket = ws
+                    await self._send_msg(ws, self._create_connect_msg())
+                    await self._recv_msg(ws)
+                    await self._send_msg(ws, self._create_subscribe_msg())
+                    await self._con_up_listener()
+
+                    self._reconnect_tries = RECONNECT_COUNT
+
+                    while not ws.closed:
+                        msg = await self._recv_msg(ws)
+                        if not msg:
+                            continue
+                        if msg.type == aiohttp.WSMsgType.ERROR:
+                            LOGGER.error("Socket message error")
+                            break
+                        if msg.type in [
+                            aiohttp.WSMsgType.CLOSE,
+                            aiohttp.WSMsgType.CLOSING,
+                            aiohttp.WSMsgType.CLOSED,
+                        ]:
+                            LOGGER.warn(
+                                f"Stopping receiving. Message type: {str(msg.type)}"
+                            )
+
+                            if (
+                                not self._auth.is_access_token_valid()
+                                or msg.data == WS_STATUS_UNAUTHORIZED
+                            ):
+                                LOGGER.debug("auth key expired, doing reauth now")
+                                await self._auth.do_auth()
+
+                            elif msg.data == WS_STATUS_GOING_AWAY:
+                                LOGGER.warn(
+                                    f"Received Going Away message: Waiting for {GOING_AWAY_DELAY} seconds"
+                                )
+                                # Give the server some time to come back up.
+                                await asyncio.sleep(GOING_AWAY_DELAY)
+
+                            break
+
+                        if msg.type != aiohttp.WSMsgType.TEXT:
+                            LOGGER.error(
+                                f"Socket message type is invalid: {str(msg.type)}"
+                            )
+                            continue
+
+                        match = RECV_MSG_MATCHER.findall(msg.data)
+                        if not match:
+                            continue
+                        self._msg_listener("{" + match[0] + "}")
+        except (
+            aiohttp.ClientConnectionError,
+            aiohttp.ClientError,
+            asyncio.TimeoutError,
+        ):
```

Comment:
This except grouping was found in multiple examples, but the aiohttp docs do not say it raises them.  I put this in due to a couple of instances where the eventsocket didn't restart and appeared to die with the "connecting to...." debug and nothing more happened - leading me to believe the connect or send/receive messages raised some error that stopped the thread.